packages/std/src/ibc.rs

```diff
@@ -391,12 +408,20 @@ impl From<IbcChannelCloseMsg> for IbcChannel {
 #[non_exhaustive]
 pub struct IbcPacketReceiveMsg {
     pub packet: IbcPacket,
+    #[cfg(feature = "ibcv3")]
+    pub relayer: Addr,
 }
 
 impl IbcPacketReceiveMsg {
+    #[cfg(not(feature = "ibcv3"))]
     pub fn new(packet: IbcPacket) -> Self {
         Self { packet }
     }
+
+    #[cfg(feature = "ibcv3")]
+    pub fn new(packet: IbcPacket, relayer: Addr) -> Self {
+        Self { packet, relayer }
+    }
```

Comment:
This is a bit of a problematic use of features. Rust assumes adding features does not break compilation. The original dependency resolver just enabled all features required by some caller and compiled the dependency once. Code like this then suddenly breaks in a hard to understand way.

But yeah, we'll probably remove all non-v3 code soon anyways.

Reply:
This is a good point. My thought was a contract is written either with one or the other. I guess for some internal tests this may cause issues, or unit tests where ibc1 contracts import ibc3 contracts.

Happy to change this if you have a better idea